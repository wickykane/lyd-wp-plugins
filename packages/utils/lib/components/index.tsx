export * from "./button";
export * from "./notice";
