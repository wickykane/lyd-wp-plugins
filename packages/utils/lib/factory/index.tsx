export * from "./context";
export * from "./ajax";
export * from "./i18n";
