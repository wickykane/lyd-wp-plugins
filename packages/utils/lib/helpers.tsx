const untrailingslashit = (str: string): string =>
    str.endsWith("/") || str.endsWith("\\") ? untrailingslashit(str.slice(0, -1)) : str;
const trailingslashit = (str: string): string => `${untrailingslashit(str)}/`;

// Allows to make an interface extension and make some properties optional (https://git.io/JeK6J)
type AllKeyOf<T> = T extends never ? never : keyof T;
type Optional<T, K> = { [P in Extract<keyof T, K>]?: T[P] };
type WithOptional<T, K extends AllKeyOf<T>> = T extends never ? never : Omit<T, K> & Optional<T, K>;

export { untrailingslashit, trailingslashit, AllKeyOf, Optional, WithOptional };
