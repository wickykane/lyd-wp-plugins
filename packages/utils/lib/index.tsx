import "setimmediate"; // Polyfill for yielding

export * from "./options";
export * from "./helpers";
export * from "./factory";
export * from "./wp-api";
export * from "./components";
