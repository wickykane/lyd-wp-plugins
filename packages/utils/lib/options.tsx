/**
 * See PHP file inc/Assets.php.
 */
abstract class BaseOptions {
    public slug: string;
    public textDomain: string;
    public version: string;
    public restUrl?: string;
    public restNamespace?: string;
    public restRoot?: string;
    public restQuery?: {};
    public restNonce?: string;
    public publicUrl?: string;

    /**
     * Convert a slug like "my-plugin" to "myPlugin". This can
     * be useful for library naming (window[""] is bad because the hyphens).
     *
     * @param slug
     * @returns
     */
    public static slugCamelCase(slug: string) {
        return slug.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
    }

    /**
     * Get the slug from the current process (webpack) instead of the PHP plugin output.
     * For some cases you need to use that.
     *
     * @param env
     * @param camelCase
     */
    public static getPureSlug(env: typeof process.env, camelCase = false) {
        return camelCase ? BaseOptions.slugCamelCase(env.slug) : env.slug;
    }
}

export { BaseOptions };
