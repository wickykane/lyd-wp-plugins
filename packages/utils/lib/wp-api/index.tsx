export * from "./rest.plugin.get";
