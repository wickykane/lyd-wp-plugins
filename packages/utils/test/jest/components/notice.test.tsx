import renderer from "react-test-renderer";
import React from "react";
import { Notice, NoticeType } from "../../../lib";
import { Provider } from "../helpers";

describe("notice", () => {
    it("should render notice with default arguments", () => {
        const tree = renderer
            .create(
                <Provider>
                    <Notice>Test</Notice>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render notice with error type", () => {
        const tree = renderer
            .create(
                <Provider>
                    <Notice type={NoticeType.Error}>Test</Notice>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render notice with info type", () => {
        const tree = renderer
            .create(
                <Provider>
                    <Notice type={NoticeType.Info}>Test</Notice>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render notice with success type", () => {
        const tree = renderer
            .create(
                <Provider>
                    <Notice type={NoticeType.Success}>Test</Notice>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
