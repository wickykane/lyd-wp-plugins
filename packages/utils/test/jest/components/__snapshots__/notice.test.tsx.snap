// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notice should render notice with default arguments 1`] = `
<div
  className="notice"
>
  <p>
    Test
  </p>
</div>
`;

exports[`notice should render notice with error type 1`] = `
<div
  className="notice notice-error"
>
  <p>
    Test
  </p>
</div>
`;

exports[`notice should render notice with info type 1`] = `
<div
  className="notice notice-info"
>
  <p>
    Test
  </p>
</div>
`;

exports[`notice should render notice with success type 1`] = `
<div
  className="notice notice-success"
>
  <p>
    Test
  </p>
</div>
`;
